src/modules/verifiable-credentials/verifiable-credentials-base.service.ts

```diff
@@ -93,12 +101,48 @@ export abstract class VerifiableCredentialsServiceBase {
     proof_options: string
   ): Promise<string>;
 
-  constructor(protected readonly _signerService: SignerService) {}
+  constructor(
+    protected readonly _signerService: SignerService,
+    private readonly _storage: VCStorageClient
+  ) {}
+
+  /**
+   * @description requests verifiable credentials of type `type`. Only EWF credentials currently supported
+   * @returns credentials query with matching verifiable presentations
+   */
+  async initiateExchange({ type, url }: ExchangeInvitation) {
+    if (type !== EWF_EXCHANGE_INVITATION) {
+      throw new Error('Only EWF exchange is supported');
+    }
+
+    const { data: vpRequest } = await axios.post<VpRequest>(url);
+    const presentationDefinitions = vpRequest.query.find(
+      (q) => q.type === VpRequestQueryType.presentationDefinition
+    )?.credentialQuery as PresentationDefinition[];
+
+    const vc: VerifiableCredential<RoleCredentialSubject>[] = [];
+    const credentials = (
+      await Promise.all(
+        presentationDefinitions.map((def) =>
+          this._storage.getCredentialsByDefinition(def)
+        )
+      )
+    ).reduce((prev, curr) => prev.concat(curr), []);
+    for (const cred of credentials) {
+      if (isRoleCredential(cred)) {
+        vc.push(cred);
+      }
+    }
```

Comment:
```suggestion
    )
    .reduce((prev, curr) => prev.concat(curr), [])
    .filter(isRoleCredential);
```
And `credentials` array is the same as `vc` should be. We get rid of one variable and 3 lines of code. What you think?